### DISCUSSION TITLE
Bagaimanakah Filter atau Sub Setting baris yg tdk dipilih ?

### DISCUSSION THREAD

**Discussion by Micoalgo** [teamalgoritma/community#591] [Q&A] [answered]
Saya mencoba melakukan Filter atau Sub Setting atas data kategorik ataupun setelah dirubah menjadi character utk id_dokter (tdp 64 level/unique) krn ada beberapa dokter yg ingin dihilangkan dari data frame disebabkan bernilai NA ketika dilakukan pemodelan step-wise forward dan both

Setelah dicoba beberapa cara a.l :
1. Menggunakan minus c (-c() ) 
cor2_oms2123 <- cor_oms2123 %>%  
  filter(id_dokter %in% -c ("174","175","345","382")) --->> cara ini menghasilkan catatan sbb ;

<img width="562" alt="image" src="https://user-images.githubusercontent.com/123436963/224607483-ec63bee6-6102-4178-8e4e-e65382b570a9.png">

2. Menggunakan Sub Setting dg operator !=

Namun setelah di cek unique dan dilakukan pemodelan kembali...id_dokter yg telah difilter tsb muncul lagi, sbb ;

<img width="672" alt="image" src="https://user-images.githubusercontent.com/123436963/224606329-34510b7a-3cfd-4b9c-9629-306b6cb91cda.png">

Bagaimanakah Filter atau Sub Setting baris yg tdk dipilih ? Terimakasih

**Comment by amaliaasf** [comment, marked as answer]
Dear Pak @Micoalgo ,

Untuk melakukan filter, sudah benar langkah yang Bapak coba di atas, yaitu bisa dengan fungsi `filter` dari library "dplyr" (cara 1) ataupun dengan subsetting menggunakan base R (cara 2).

Namun ada sedikit koreksi dari code-nya yang mengakibatkan hasil running code error dan tidak sesuai dengan yang diharapkan.

1. Code untuk menggunakan fungsi filter(): 

```
cor_oms2123 %>%
filter(!id_dokter %in% c("174","175","345","382"))
```
_Note: Bapak bisa menambahkan tanda `!` pada id_dokter untuk melakukan negasi._

2. Code untuk base R:

`cor_oms2123[cor_oms2123$id_dokter != "174" & cor_oms2123$id_dokter != "175" & cor_oms2123$id_dokter != "345" & cor_oms2123$id_dokter != "382", ]`

_Note: Operator kondisi yang tepat adalah & (AND). Karena kalau operatornya | (OR) ketika salah satu kondisi terpenuhi maka output akan tertampilkan._

Demikian yang dapat kami sampaikan, semoga menjawab! ^^

**Reply by Micoalgo** [reply]
Terimakasih jawabannya
